gitindex/index.go

```diff
@@ -532,6 +528,248 @@ func newIgnoreMatcher(tree *object.Tree) (*ignore.Matcher, error) {
 	return ignore.ParseIgnoreFile(strings.NewReader(content))
 }
 
+type buildOutput struct {
```

Comment:
nit: I am not sure about the name, but I also cannot come up with a better one :-). IMO we could just return the individual parameters instead of the struct, especially because `buildOutput` doesn't have any meaning beyond collecting return values. This would also solve the issue that `prepareNormalBuild` always returns an empty `changeOrDeletedPaths`.